layers/meta-resin-allwinner/recipes-kernel/linux/linux-mainline_4.15.18.bb

```diff
@@ -0,0 +1,25 @@
+SECTION = "kernel"
+DESCRIPTION = "Mainline Linux kernel"
+LICENSE = "GPLv2"
+LIC_FILES_CHKSUM = "file://COPYING;md5=d7810fab7487fb0aad327b76f1be7cd7"
+COMPATIBLE_MACHINE = "(sun4i|sun5i|sun7i|sun8i|sun50i)"
+
+inherit kernel
+
+# Since we're not using git, this doesn't make a difference, but we need to fill
+# in something or kernel-yocto.bbclass will fail.
+KBRANCH ?= "master"
+
+# Pull in the devicetree files into the rootfs
+RDEPENDS_${KERNEL_PACKAGE_NAME}-base += "kernel-devicetree"
+
+KERNEL_EXTRA_ARGS += "LOADADDR=${UBOOT_ENTRYPOINT}"
+
+S = "${WORKDIR}/linux-${PV}"
+	
+SRC_URI[md5sum] = "2c7a9404ddf29580f14c61eb87e50c6e"
+
+SRC_URI = "https://cdn.kernel.org/pub/linux/kernel/v4.x/linux-${PV}.tar.xz \
```

Comment:
Why not use https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux-stable.git/ ?

Reply:
This recipe was inspired from the way Sunxi did it in the BSP layer. The kernel can be pulled the way you specify, but I don't see the problem with pulling the kernel this way. In fact, it's faster to pullthe kernel this way because you're not pulling all the huge git history.